hardware/Meteorologisk.cpp

```diff
@@ -48,22 +48,90 @@ std::string ReadFile(std::string filename)
 }
 #endif
 
+#define Meteorologisk_URL "https://api.met.no/weatherapi/locationforecast/2.0/complete.json"
+#define Meteorologisk_forecast_URL "https://darksky.net/#/f/"
+
 CMeteorologisk::CMeteorologisk(const int ID, const std::string &Location) :
 	m_Location(Location)
 {
 	m_HwdID=ID;
-	Init();
+	m_URL = "";
 }
 
 void CMeteorologisk::Init()
 {
+	std::stringstream sURL;
+	m_Lat = 1;
+	m_Lon = 1;
+
+	// Try the hardware location first, then fallback on the global settings
+	if(!m_Location.empty())
+	{
+		std::vector<std::string> strarray;
+		if(m_Location != "0")
+		{
+			StringSplit(m_Location, ",", strarray);
+			if(strarray.size() == 2)
+			{
+				m_Lat = strtod(strarray[0].c_str(),nullptr);
+				m_Lon = strtod(strarray[1].c_str(),nullptr);
+			}
+			else
+			{
+				StringSplit(m_Location, ";", strarray);
+				if(strarray.size() == 2)
+				{
+					m_Lat = strtod(strarray[0].c_str(),nullptr);
+					m_Lon = strtod(strarray[1].c_str(),nullptr);
+					Log(LOG_STATUS, "Found Location with semicolon as seperator instead of comma.");
+				}
+			}
+		}
+		else
+		{
+			int nValue;
+			std::string sValue;
+			if (!m_sql.GetPreferencesVar("Location", nValue, sValue))
+			{
+				Log(LOG_ERROR, "Invalid Location found in Settings! (Check your Latitude/Longitude!)");
+			}
+			else
+			{
+				StringSplit(sValue, ";", strarray);
+				if(strarray.size() == 2)
+				{
+					m_Lat = strtod(strarray[0].c_str(),nullptr);
+					m_Lon = strtod(strarray[1].c_str(),nullptr);
+					Log(LOG_STATUS, "Using Domoticz default location from Settings as Location!");
+				}
+			}
+		}
+	}
+
+	if(!(m_Lat == 1 || m_Lon == 1))
+	{
+		sURL << Meteorologisk_URL << "?lat=" << m_Lat << "&lon=" << m_Lon;
+		m_URL = sURL.str();
+	}
+
+	std::string szLoc = CURLEncode::URLEncode(m_Location);
+	sURL.flush();
```

Comment:
best to use this to clear/reset a stringstream

std::stringstream().swap(sUrl);

Reply:
used str(std::string()) and clear()

Is that ok too?

Reply:
Yes, but the swap is c++11 prefered way